src/engine/tile/tileset.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <external/SDL2/include/SDL.h>
+#include <stdio.h>
+
+struct Tileset 
+{
```

Comment:
Maybe use a SpriteSheet + m_Bitmask to hold the data for tileset? What do you think?

Reply:
rather than having 
```
  Uint32 tile_size;
    Uint32 n_rows;
    Uint32 n_cols;
    ```
    inside this struct, can be
    ```
    SpriteSheet sprite_sheet
    std::vector<Uint8> bitmasks
    ```